tests/operators/test_agnostic_aggregate_check.py

```diff
@@ -33,253 +36,213 @@
 
 import astro.sql as aql
 from astro.sql.table import Table
-from tests.operators.utils import DEFAULT_SCHEMA
+from tests.operators.utils import DEFAULT_SCHEMA, get_dag, get_table_name, run_dag
 
 log = logging.getLogger(__name__)
 DEFAULT_DATE = timezone.datetime(2016, 1, 1)
+CWD = pathlib.Path(__file__).parent
+
+DATABASES_LIST = ["postgres", "bigquery", "sqlite", "snowflake"]
+
+
+@pytest.fixture(scope="session")
+def tables(request):
+    aggregate_table = Table(
+        "aggregate_check_test",
+        database="pagila",
+        conn_id="postgres_conn",
+        schema="airflow_test_dag",
+    )
+    aggregate_table_bigquery = Table(
+        "aggregate_check_test",
+        conn_id="bigquery",
+        schema=DEFAULT_SCHEMA,
+    )
+    aggregate_table_sqlite = Table("aggregate_check_test", conn_id="sqlite_conn")
+    aggregate_table_snowflake = Table(
+        table_name=get_table_name("aggregate_check_test"),
+        database=os.getenv("SNOWFLAKE_DATABASE"),  # type: ignore
+        schema=os.getenv("SNOWFLAKE_SCHEMA"),  # type: ignore
+        warehouse=os.getenv("SNOWFLAKE_WAREHOUSE"),  # type: ignore
+        conn_id="snowflake_conn",
+    )
+    path = str(CWD) + "/../data/homes_merge_1.csv"
+    tables = {
+        "postgres": aggregate_table,
+        "bigquery": aggregate_table_bigquery,
+        "sqlite": aggregate_table_sqlite,
+        "snowflake": aggregate_table_snowflake,
+    }
+    for _, table in tables.items():
+        aql.load_file(
+            path=path,
+            output_table=table,
+        ).operator.execute({"run_id": "foo"})
 
+    yield tables
 
-class TestAggregateCheckOperator(unittest.TestCase):
-    """
-    Test Postgres Merge Operator.
-    """
 
-    cwd = pathlib.Path(__file__).parent
+@pytest.mark.parametrize("database", DATABASES_LIST)
+def test_exact_value(sample_dag, database, tables):
+    @aql.transform
+    def get_table(input_table: Table):
+        return "SELECT * FROM {{input_table}}"
 
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-        cls.aggregate_table = Table(
-            "aggregate_check_test",
-            database="pagila",
-            conn_id="postgres_conn",
-            schema="airflow_test_dag",
-        )
-        cls.aggregate_table_bigquery = Table(
-            "aggregate_check_test",
-            conn_id="bigquery",
-            schema=DEFAULT_SCHEMA,
+    with sample_dag:
+        aggregate_table = get_table(tables[database])
+        aql.aggregate_check(
+            table=aggregate_table,
+            check="select count(*) FROM {{table}}",
+            greater_than=4,
+            less_than=4,
```

Comment:
It would be great to have some sort of assert.  Do we expect this task to log something? Does it add anything Xcom?

Reply:
The 'aggregate_check' op will raise a value error,  so we expect the dag to fail if the test fails.